DuckDuckGo/TabViewController.swift

```diff
@@ -925,11 +967,43 @@ extension TabViewController: WKNavigationDelegate {
         }
     }
     
+    /*
+     Some files might be previewed by webkit but in order to share them
+     we need to download them first.
+     This method stores the temporary download or clears it if necessary
+     */
+    private func setupOrClearTemporaryDownload(for navigationResponse: WKNavigationResponse) {
+        let downloadManager = AppDependencyProvider.shared.downloadsManager
+        
+        if let downloadMetaData = downloadManager.downloadMetaData(for: navigationResponse), !downloadMetaData.mimeType.isHTML {
+            let cookieStore = webView.configuration.websiteDataStore.httpCookieStore
+            temporaryDownload = downloadManager.setupDownload(navigationResponse,
+                                                              cookieStore: cookieStore,
+                                                              temporary: true)
+        } else {
+            temporaryDownload = nil
+        }
+    }
+    
     func webView(_ webView: WKWebView,
                  decidePolicyFor navigationResponse: WKNavigationResponse,
                  decisionHandler: @escaping (WKNavigationResponsePolicy) -> Void) {
-        decisionHandler(.allow)
-        url = webView.url
+      
+        guard !navigationResponse.canShowMIMEType else {
+            setupOrClearTemporaryDownload(for: navigationResponse)
+            url = webView.url
+            decisionHandler(.allow)
+            return
+        }
+       
+        let downloadManager = AppDependencyProvider.shared.downloadsManager
+        let cookieStore = webView.configuration.websiteDataStore.httpCookieStore
+        
+        if let download = downloadManager.setupDownload(navigationResponse, cookieStore: cookieStore) {
+            downloadManager.startDownload(download)
```

Comment:
I think that according to the design we need to prompt the user for the download to start.

Reply:
Yep, this is left out intentionally since I didn't want to change anything related to the UI on this PR besides the file preview. I'll handle this prompt together on the notification toast PR that will follow